new/detector/implementation/javascript/string/string.go

```diff
@@ -0,0 +1,83 @@
+package string
+
+import (
+	"fmt"
+
+	"github.com/bearer/curio/new/detector/types"
+	"github.com/bearer/curio/new/language/tree"
+	"github.com/bearer/curio/pkg/util/stringutil"
+
+	generictypes "github.com/bearer/curio/new/detector/implementation/generic/types"
+	languagetypes "github.com/bearer/curio/new/language/types"
+)
+
+type stringDetector struct{}
+
+func New(lang languagetypes.Language) (types.Detector, error) {
+	return &stringDetector{}, nil
+}
+
+func (detector *stringDetector) Name() string {
+	return "string"
+}
+
+func (detector *stringDetector) DetectAt(
+	node *tree.Node,
+	evaluator types.Evaluator,
+) ([]interface{}, error) {
+	switch node.Type() {
+	case "string":
+		return []interface{}{generictypes.String{Value: stringutil.StripQuotes(node.Content())}}, nil
+	case "template_string":
+		return handleTemplateString(node, evaluator)
+	case "binary_expression":
+		if node.AnonymousChild(0).Content() == "+" {
+			return concatenateChildren(node, evaluator)
+		}
+	}
+
+	return nil, nil
+}
+
+func concatenateChildren(node *tree.Node, evaluator types.Evaluator) ([]interface{}, error) {
+	value := ""
+
+	for i := 0; i < node.ChildCount(); i += 1 {
+		child := node.Child(i)
+		if !child.IsNamed() {
+			continue
+		}
+
+		detections, err := evaluator.ForNode(child, "string", true)
+		if err != nil {
+			return nil, err
+		}
+
+		switch len(detections) {
+		case 0:
+			value += "*"
+		case 1:
+			value += detections[0].Data.(generictypes.String).Value
+		default:
+			return nil, fmt.Errorf("expected single string detection but got %d", len(detections))
+		}
+	}
+
+	return []interface{}{generictypes.String{Value: value}}, nil
+}
+
+func handleTemplateString(node *tree.Node, evaluator types.Evaluator) ([]interface{}, error) {
+	text := ""
+
+	err := node.EachPart(func(partText string) error {
+		text += partText
+		return nil
+	}, func(child *tree.Node) error {
+		text += "*"
```

Comment:
When the child is `template_substitution`, I think we should take it's named child (`child.Child(1)`) and use similar logic to the concatenation case: 
https://github.com/Bearer/curio/pull/452/files#diff-7bec69bed8205114a204bc68a1a8a822867aa06d1ca446b3f24276bfb5cccd00R51-R63